### ISSUE TITLE
AttributeError causes internal error for types defined in other scope than where the error was raised

### ISSUE THREAD

**Issue by MrGreenTea** [aroberge/friendly#212]
**Describe the bug**

**To Reproduce**
1. Create `mod.py`:
```py
class A:
  pass


a = A()
```
2. use in `test.py`:
```py
from mod import a


a.no_attr
```
3. run with `friendly`: `python -m friendly test.py`
4. get an internal error message
```
╭──────────────────────────────── Traceback ─────────────────────────────────╮
│ Traceback (most recent call last):                                         │
│   File "test.py", line 4, in <module>                                      │
│     a.no_attr                                                              │
│ AttributeError: 'A' object has no attribute 'no_attr'                      │
│                                                                            │
│ An AttributeError occurs when the code contains something like object.x    │
│ and x is not a method or attribute (variable) belonging to object.         │
│                                                                            │
│ Internal error for Friendly-traceback. Please report this example to       │
│ https://github.com/aroberge/friendly/issues                                │
│                                                                            │
│ Exception raised on line 4 of file 'test.py'.                              │
│                                                                            │
│        2:                                                                  │
│        3:                                                                  │
│     -->4: a.no_attr                                                        │
│                                                                            │
│     a:  <mod.A object>                                                     │
╰────────────────────────────────────────────────────────────────────────────╯
```

**Expected behavior**
Show something about type `A`.

**Which Python version and Friendly version are you using**
Python: 3.9.5
friendly: 0.3.53

This seems to be because of the logic in `friendly.info_variables.get_object_from_name` where it only shows in the locals and globals of that frame where the error was raised. This does not find the type in `mod.py` and returns `None`.

**Comment by MrGreenTea** [comment]
Can also be reproduced with a single file:
```py
def f():
    class A: pass
    return A()

a = f()
a.no_attr
```

**Comment by aroberge** [comment]
Thanks for the report. I'll work on it right away.

**Comment by aroberge** [comment]
This should be fixed in version 0.3.54 uploaded to Pypi.  Unfortunately, some more complex cases can occur when the correct object cannot be identified, currently raising a similar internal error.  For example:
```python
class A:
    pass
def f():
    class A: pass
    return A()

a = f()
a.no_attr
```

will give rise to
```python
AttributeError: 'A' object has no attribute 'no_attr'
```
In trying to find which object `A` refers to, the local one is found but it does not correspond to the object of which `a` is an instance.

**Comment by MrGreenTea** [comment]
@aroberge Thanks for the insanely quick fix!
